opentelemetry-api/src/opentelemetry/util/_importlib_metadata.py

```diff
@@ -12,27 +12,163 @@
+# points for 3.7 every time the entry points function is called.
+result_entry_points_37 = {}
 
 # FIXME remove this when support for 3.7 is dropped.
 if version_info.minor == 7:
     # pylint: disable=import-error
-    from importlib_metadata import entry_points, version  # type: ignore
+    from importlib_metadata import EntryPoint
+    from importlib_metadata import (
+        entry_points as importlib_metadata_entry_points,
+    )
+    from importlib_metadata import version
+
+    def entry_points(
+        group: str = None, name: str = None
+    ) -> Union[Tuple, Dict]:
+
+        if group is None and name is None:
+
+            if not result_entry_points_37:
+
+                for entry_point in importlib_metadata_entry_points():
+
+                    if entry_point.group not in result_entry_points_37.keys():
+                        result_entry_points_37[entry_point.group] = []
+
+                    (
+                        result_entry_points_37[entry_point.group].append(
+                            entry_point
+                        )
+                    )
+
+                for key, value in result_entry_points_37.items():
+                    result_entry_points_37[key] = tuple(value)
+
+            return result_entry_points_37
+
+        if group is not None and name is None:
+            return tuple(
+                entry_point
+                for entry_point in importlib_metadata_entry_points(group=group)
+            )
+
+        if group is None and name is not None:
+            return tuple(
+                entry_point
+                for entry_point in importlib_metadata_entry_points(name=name)
+            )
+
+        return tuple(
+            entry_point
+            for entry_point in importlib_metadata_entry_points(
+                group=group, name=name
+            )
+        )
+
+    __all__ = [
+        "entry_points",
+        "EntryPoint",
+        "version",
+    ]
 
 # FIXME remove this file when support for 3.9 is dropped.
 elif version_info.minor in (8, 9):
     # pylint: disable=import-error
+    from importlib.metadata import EntryPoint
     from importlib.metadata import (
         entry_points as importlib_metadata_entry_points,
     )
     from importlib.metadata import version
 
-    def entry_points(group: str, name: str):  # type: ignore
-        for entry_point in importlib_metadata_entry_points()[group]:
-            if entry_point.name == name:
-                yield entry_point
+    def entry_points(
+        group: str = None, name: str = None
+    ) -> Union[Tuple, Dict]:
+
+        result_key_entry_points = importlib_metadata_entry_points()
+
+        if group is None and name is None:
+            return result_key_entry_points
+
+        if group is not None and group not in result_key_entry_points.keys():
+            return tuple()
+
+        if group is not None and name is None:
+            return result_key_entry_points[group]
```

Comment:
```suggestion
            return tuple(result_key_entry_points[group])
```

Reply:
`result_key_entry_points[group]` is a tuple already